docs/Reference-for-template.json.md

```diff
@@ -1,55 +1,101 @@
-|tags|You can use tags to improve the metadata of your project. To specify the project language you have to add the tag `language`, and if you want to make your project searchable via the `dotnet new --type` command you have to use the tag `type`|
-|shortName|A default shorthand for selecting the template (applies to environments where the template name is specified by the user - not selected via a GUI). this is the name shown as Short Name in `dotnet new` list of templates, and is the name to use command list to run this template |
-|postActions|Enables actions to be performed after the project is created. See https://github.com/dotnet/templating/wiki/Post-Action-Registry|
+### Template Definition
+|Name|Description|Mandatory|
+|---|---|---|
+|`identity`|A unique name for this template|yes|
+|`author`|The author of the template|no|
+|`classifications`|Zero or more characteristics of the template which may be used in search. In this field you define the values shown as Tags in `dotnet new`|no|
+|`name`|The name for the template. This is displayed as the template name when using `dotnet new` and Visual Studio.|yes|
+|`groupIdentity`|The ID of the group this template belongs to. This allows multiple templates to be displayed as one, with the the decision for which one to use based on the template options. |no|
+|`tags`|You can use tags to improve the metadata of your project. Well-known tags are: `language` and `type`. To specify the template language, use the tag `language`. To specify the template type, use the tag `type`. Supported types are: `project`, `item`, `solution`.|no|
+|`shortName`|A name for selecting the template in CLI environment. This is the name shown as `Short Name` in `dotnet new` list of templates, and is the name to use to run this template. |yes|
+|`postActions`|Enables actions to be performed after the project is created. See [the article](Post-Action-Registry.md) on post actions for more details.|no|
+|`constraints`|[Template constraints](#template-constraints)||no|
+
+#### Template constraints
+
+Available since .NET SDK 7.0.100.
+The template may define the constraints all of which must be met in order for the template to be used. 
+In case constraints are not met, the template will be installed, however will not be visible by default. 
+For the details on available constraints, refer to [the article](Constraints.md).
+The constraints are defined under `constraints` property (top level in template.json). `constraints` contains objects (constraint definition). Each constraint should have a unique name, `type` and optional arguments (`args`). Argument syntax depends on the constraint implementation.
+
+```json
+"constraints": {
+    "linux-only": {
+        "type": "os",
+        "args": "Linux"
+    },
+    "sdk-only": {
+        "type": "host",
+        "args": [
+            {
+                "hostname": "dotnetcli",
+                "version": "[6.0.100, )"
+            }
+        ]
+    }
+}
+```  
```

Comment:
```suggestion
```json
"constraints": {
  "linux-only": {
    "type": "os",
    "args": "Linux"
  },
  "sdk-only": {
    "type": "host",
    "args": [
      {
        "hostname": "dotnetcli",
        "version": "[6.0.100, )"
      }
    ]
  }
}
```  
```

We have idention of 2 spaces in other json examples